src/abstractdataframe/reshape.jl

```diff
@@ -420,40 +456,58 @@ d1m = meltdf(d1, [:a, :b, :e])
 ```
 
 """
-function stackdf(df::AbstractDataFrame, measure_vars::Vector{Int}, id_vars::Vector{Int})
+function stackdf(df::AbstractDataFrame, measure_vars::Vector{Int},
+                 id_vars::Vector{Int}; measure_name::Symbol=:variable,
+                 value_name::Symbol=:value)
     N = length(measure_vars)
     cnames = names(df)[id_vars]
-    insert!(cnames, 1, :value)
-    insert!(cnames, 1, :variable)
+    insert!(cnames, 1, value_name)
+    insert!(cnames, 1, measure_name)
     DataFrame(Any[RepeatedVector(_names(df)[measure_vars], nrow(df), 1),   # variable
                   StackedVector(Any[df[:,c] for c in measure_vars]),     # value
                   [RepeatedVector(df[:,c], 1, N) for c in id_vars]...],     # id_var columns
               cnames)
 end
-function stackdf(df::AbstractDataFrame, measure_var::Int, id_var::Int)
-    stackdf(df, [measure_var], [id_var])
+function stackdf(df::AbstractDataFrame, measure_var::Int, id_var::Int;
+                 measure_name::Symbol=:variable, value_name::Symbol=:value)
+    stackdf(df, [measure_var], [id_var]; measure_name=measure_name,
+            value_name=value_name)
 end
-function stackdf(df::AbstractDataFrame, measure_vars, id_var::Int)
-    stackdf(df, measure_vars, [id_var])
+function stackdf(df::AbstractDataFrame, measure_vars, id_var::Int;
+                 measure_name::Symbol=:variable, value_name::Symbol=:value)
```

Comment:
The annotation is only useful to ensure an error is printed immediately rather than later when the code attempts doing something which only works for symbols. In general it's better to strictly type arguments when it makes sense for clarity, and it makes the documentation clearer.